### PR TITLE
Add playlist support.

### PR DESCRIPTION
**Pull request by ccxcz**
This addresses being able to add full discs / albums or any other unit atomically as a sequence (issue #8). It also allows for adding items with correct metadata when those are stored in the playlist (such as cue sheet).

### CODE REVIEW

File: src/ashuffle.c [hunk 1]
```diff
@@ -161,7 +161,22 @@ int build_songs_mpd(struct mpd_connection * mpd,
...
-    if (mpd_run_add(mpd, shuffle_pick(songs)) != true) { mpd_perror(mpd); }
+    const char * item = shuffle_pick(songs);
+    const char * suffix = item;
+    /* find last dot */
+    for(int n = 0; item[n]; n++) {
+        if(item[n] == '.') {
+            suffix = &item[n+1];
+        }
+    }
+    bool is_playlist = ( \
```
Comment (joshkunz) [inline src/ashuffle.c]: No need for the ternary operator, just use logical or between the `strcmp(...) == 0` checks.

**Comment by joshkunz** [comment]
Thanks for the pull request!

So just to clarify, this only works with `--nocheck -f` when the user supplies a list of paths to playlist files?

File: src/ashuffle.c (hunk 1 above)
Comment (ccxcz) [reply src/ashuffle.c]: That would result in superfluous checks once we determine this is actually a playlist from the suffix.

**Comment by ccxcz** [comment]
Yes. While it's technically possible to add extra condition on URI suffix to check if file exists via listfiles command rather than querying the medialib to mirror the behaviour of queue_random_song this works well enough for me as is.

File: src/ashuffle.c (hunk 1 above)
Comment (joshkunz) [reply src/ashuffle.c]: Logical OR (`||`) is short-cutting in C (as in most languages), so no superfluous checks will take place.

**Comment by joshkunz** [comment]
Agreed this seems reasonable enough to me. Just wanted to clarify. If #8 ever gets implemented, we should fold this functionality in with that (whatever it is). This is definitely a bit of hack, but with `--nocheck -f` we kind of assume you know what you're doing.

Please update the documentation for `--nocheck` to mention this behavior.

**Comment by joshkunz** [comment]
This PR has not been updated in ~1yr. Closing due to it now being very out of date, and inactive.

**Pull request closed without merging**